### PR TITLE
Add support for IPGroups in Firewall Policy Rules

### PR DESCRIPTION
**Pull request by tejasshah7**
## Description

This change adds support for IPGroups in Firewall Policy Application, Network and NAT rules. Here is the commandlet review PR: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/606

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ x The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md [hunk 1]
```diff
@@ -138,6 +154,21 @@ Accept pipeline input: False
...
 
+### -SourceIpGroup
+The source ipgroups of the rule
+
+```yaml
+Type: String[]
+Parameter Sets: (All)
+Aliases:
+
+Required: True
```
Comment (VeryEarly) [inline src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md:165]: this should be false, can you fix it manually

File: src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md [hunk 2]
```diff
@@ -77,6 +78,21 @@ Accept pipeline input: False
...
 
+### -DestinationIpGroup
+The destination ipgroups of the rule
+
+```yaml
+Type: String[]
+Parameter Sets: (All)
+Aliases:
+
+Required: True
```
Comment (VeryEarly) [inline src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md:89]: this should be false, can you fix it manually

File: src/Network/Network/help/New-AzFirewallPolicyNatRule.md [hunk 3]
```diff
@@ -139,6 +139,21 @@ Accept pipeline input: False
...
 
+### -SourceIpGroup
+The source ipgroups of the rule
+
+```yaml
+Type: String[]
+Parameter Sets: (All)
+Aliases:
+
+Required: True
```
Comment (VeryEarly) [inline src/Network/Network/help/New-AzFirewallPolicyNatRule.md:150]: this should be false, can you fix it manually

**Comment by VeryEarly** [comment]
please also resolve the merge conflicts

File: src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md (hunk 1 above)
Comment (tejasshah7) [reply src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md:165]: Fixed

File: src/Network/Network/help/New-AzFirewallPolicyNatRule.md (hunk 3 above)
Comment (tejasshah7) [reply src/Network/Network/help/New-AzFirewallPolicyNatRule.md]: Fixed

File: src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md (hunk 2 above)
Comment (tejasshah7) [reply src/Network/Network/help/New-AzFirewallPolicyNetworkRule.md:89]: Fixed

**Comment by tejasshah7** [comment]
> please also resolve the merge conflicts

Done

**Comment by number213** [comment]
@VeryEarly please merge if everything's okay

**Pull request merged**